File: matrix.exponential/python/matrix.exponential.v6.py
from matexp import matexp
import numpy as np
import sys
import os
from datetime import datetime

pid = os.getpid()
print("pid is : ",pid)
now = datetime.now()
print("start time : ",now)  


n = int(sys.argv[1])
m = int(sys.argv[2])
s = int(sys.argv[3])
stem = str(sys.argv[4])

np.random.seed(s)

X = np.random.rand(n,n)
Y = matexp(X,m)


fileX = stem + ".X.n=" + str(n) + ".m=" + str(m) + ".s=" + str(s)
fileY = stem + ".Y.n=" + str(n) + ".m=" + str(m) + ".s=" + str(s)


np.save(fileX,X)
np.save(fileY,Y)

now = datetime.now()
print("end time : ",now) 

File: matrix.exponential/R/matrix.exponential.v6.R

source("matrix.exponential.R")

args <- commandArgs(trailingOnly=TRUE)

n <- as.numeric(args[1])
m <- as.numeric(args[2])
s <- as.numeric(args[3])
stem <- args[4]

pid <- as.character(Sys.getpid())
now <- as.character(Sys.time())

cat("pid i : ",pid,'\n')
cat("start time : ",now,'\n')


set.seed(s)

X <- matrix(runif(n^2),n,n)
Y <- matexp(X,m)


fileX <- paste(stem,".X.n=",n,".m=",m,".s=",s,".RData",sep="")
fileY <- paste(stem,".Y.n=",n,".m=",m,".s=",s,".RData",sep="")


save(X,file=fileX)
save(Y,file=fileY)


now <- as.character(Sys.time())
cat("end time : ",now,'\n')
